exec/cnex/stack.c

```diff
@@ -28,6 +28,7 @@ TStack *createStack(int capacity)
 
 void destroyStack(TStack *stack)
 {
+    assert(isEmpty(stack));
     while (!isEmpty(stack)) {
```

Comment:
That assert wasn't always there.  It was commented out.  Which is better?  The assert, or the loop to clear the stack, which.../should/ already be empty--but might not be?

Reply:
The assert is better, because you're saying "if we get here, then the stack should be empty". Otherwise, there's a bug somewhere else. What's the worst that can happen if your program logic is broken and the assertion doesn't hold? A memory leak. If your logic is broken then you've got bigger problems than a memory leak.